components/style/values/specified/image.rs

```diff
@@ -626,6 +630,25 @@ impl GenericsLineDirection for LineDirection {
                 if compat_mode == CompatMode::Modern => true,
             LineDirection::Vertical(Y::Top)
                 if compat_mode != CompatMode::Modern => true,
+            #[cfg(feature = "gecko")]
+            LineDirection::MozPosition(Some(LegacyPosition {
+                horizontal: ref x,
+                vertical: ref y,
+            }), None) => {
+                let x = match *x {
+                    OriginComponent::Center => true,
+                    OriginComponent::Length(LengthOrPercentage::Percentage(ComputedPercentage(val)))
+                     if val == 0.5 => true,
```

Comment:
Please indent this in a less confusing way.

Why not:

```rust
OriginComponent::Length(LengthOrPercentage::Percentage(ComputedPercentage(val))) => {
    val == 0.5
}
```

Ideally with a comment regarding it being the default and so on.